code/modules/mob/living/carbon/carbon.dm

```diff
@@ -855,6 +855,7 @@
 		var/obj/item/I = X
 		I.acid_level = 0 //washes off the acid on our clothes
 		I.extinguish() //extinguishes our clothes
+		I.cut_overlay(GLOB.fire_overlay, TRUE)

```

Comment:
.extinguish() already call cut_overlay(GLOB.fire_overlay, TRUE)
It really need be doubled here?

Reply:
This one does need to be doubled. Without this cut_overlay, the fire overlay stays forever when there's mixed overlays (Acid, Fire, and/or Damage overlays).